### ISSUE TITLE
make the HCM settings available at the VS level, not Gateway

### ISSUE THREAD

**Issue by bcollard** [solo-io/gloo#5549]
_Currently, HCM settings are managed in the `Gateway` resource. Different Virtual Services, which are bound to the same Gateway, may require different HCM settings. I considered creating custom gateways, but this seems like a problematic approach. If my understanding is correct, gateways define listeners. So it's implied that port settings would need to be adjusted. At a large scale of virtual services with unique HCM settings, this can be problematic. [...] Does it make sense to support per-VS HCM settings?_

Internal reference: https://solo-io-corp.slack.com/archives/CEDCS8TAP/p1636383388103000

**Comment by sam-heilbron** [comment]
Adding a comment here to clarify the intent. I imagine the request is to allow a single listener to support multiple filter chains, each with a unique HCM. Currently, a Gloo Gateway produces an Envoy listener and HttpGateways can only be configured with a single HCM listener. Regardless of implementation details, the desire is to have a single listener support multiple filter chains, each with distinct HCMs. Can you confirm that?

**Comment by bcollard** [comment]
From my understanding, the goal is to use multiple VS, selected by one single Gateway (envoy listener). Each VS might have different HCM settings.
I agree #5324 will lay the foundation for this feature request.
@guydaichs feel free to add a comment if needed

**Comment by sam-heilbron** [comment]
The HttpConnectionManager settings have historically lived at the Gateway level, specifically the HttpGateway (https://docs.solo.io/gloo-edge/latest/guides/traffic_management/listener_configuration/http_connection_manager/). It proved too challenging to introduce this concept to the VirtualService (since VirtualServices own routing configuration, and less filter configuration). Therefore, to expose this funtionality of multiple HCMs and keep the functionality of application teams being able to own VirtualServices, we introduced a new resource that is designed to be maintained by application teams and provide this functionality.

We recently introduced a HybridGateway (https://docs.solo.io/gloo-edge/latest/guides/traffic_management/listener_configuration/hybrid_gateway/) which allowed for unique filter chains (with different HCMs) on a single listener. The downside was that these defintions all lived on the Gateway and thus were managed by a single team. In our most recent work, we've enabled the HttpGateway to be defined in a separate resource (managed by application team) and the top level Gateway resource can select that HttpGateway either via ref selection or label selection.

This new feature will land in the latest 1.11 beta and 1.10.8 of open source Gloo. Gloo Enterprise will support this in the latest 1.11 beta and 1.10.5.
